Add filtering on the notebook's test jobs

atheo89 opened the pull request:
Implement test job filtering in the notebook's CI pipeline to ensure that only the altered image triggers the pipeline during development.
Related to: https://github.com/opendatahub-io/notebooks/issues/102

harshad16 commented on ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:
```diff
@@ -178,7 +178,7 @@ tests:
     workflow: opendatahub-io-ci-image-mirror
 - as: notebook-jupyter-minimal-ubi8-python-3-8-pr-image-mirror
-  skip_if_only_changed: ^docs/|\\.(md)$|^(\.gitignore|LICENSE|OWNERS|PROJECT)$
+  run_if_changed: ^(jupyter/minimal/ubi8-python-3.8/*)
```
```suggestion
  run_if_changed: ^(base/ubi8-python-3.8/*)|(jupyter/minimal/ubi8-python-3.8/*)
```
Shall we do something like to emulate, changes if the base image build ?

harshad16 commented on ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:
```diff
@@ -218,7 +218,7 @@ tests:
     workflow: opendatahub-io-ci-image-mirror
 - as: notebook-jupyter-pytorch-ubi8-python-3-8-pr-image-mirror
-  skip_if_only_changed: ^docs/|\\.(md)$|^(\.gitignore|LICENSE|OWNERS|PROJECT)$
+  run_if_changed: ^(base/ubi8-python-3.8/*)|(jupyter/minimal/ubi8-python-3.8/*)|(jupyter/pytorch/ubi8-python-3.8/*)
```
```suggestion
  run_if_changed: ^(base/ubi8-python-3.8/*)|(jupyter/minimal/ubi8-python-3.8/*)|(jupyter/datascience/ubi8-python-3.8/*)|(jupyter/pytorch/ubi8-python-3.8/*)
```

harshad16 commented on ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:
```diff
@@ -238,7 +238,7 @@ tests:
     workflow: opendatahub-io-ci-image-mirror
 - as: notebook-cuda-ubi8-python-3-8-pr-image-mirror
-  skip_if_only_changed: ^docs/|\\.(md)$|^(\.gitignore|LICENSE|OWNERS|PROJECT)$
+  run_if_changed: ^(cuda/ubi8-python-3.8/*)
```
```suggestion
  run_if_changed: ^(base/ubi8-python-3.8/*)|(cuda/ubi8-python-3.8/*)
```

harshad16 commented on ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:
```diff
@@ -391,7 +391,7 @@ tests:
     workflow: opendatahub-io-ci-image-mirror
 - as: runtime-cuda-tensorflow-ubi8-python-3-8-pr-image-mirror
-  skip_if_only_changed: ^docs/|\\.(md)$|^(\.gitignore|LICENSE|OWNERS|PROJECT)$
+  run_if_changed: ^(base/ubi8-python-3.8/*)|(runtimes/tensorflow/ubi8-python-3.8/*)
```
```suggestion
  run_if_changed: ^(base/ubi8-python-3.8/*)|(cuda/ubi8-python-3.8/*)|(runtimes/tensorflow/ubi8-python-3.8/*)
```

harshad16 commented on ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:
```diff
@@ -490,7 +490,7 @@ tests:
     workflow: opendatahub-io-ci-image-mirror
 - as: notebook-cuda-ubi9-python-3-9-pr-image-mirror
-  skip_if_only_changed: ^docs/|\\.(md)$|^(\.gitignore|LICENSE|OWNERS|PROJECT)$
+  run_if_changed: ^(cuda/ubi9-python-3.9/*)
```
```suggestion
  run_if_changed: ^(base/ubi9-python-3.9/*)|(cuda/ubi9-python-3.9/*)
```

harshad16 commented on ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:
```diff
@@ -642,7 +642,7 @@ tests:
     workflow: opendatahub-io-ci-image-mirror
 - as: runtime-cuda-tensorflow-ubi9-python-3-9-pr-image-mirror
-  skip_if_only_changed: ^docs/|\\.(md)$|^(\.gitignore|LICENSE|OWNERS|PROJECT)$
+  run_if_changed: ^(base/ubi9-python-3.9/*)|(runtimes/tensorflow/ubi9-python-3.9/*)
```
```suggestion
  run_if_changed: ^(base/ubi9-python-3.9/*)|(cuda/ubi9-python-3.9/*)|(runtimes/tensorflow/ubi9-python-3.9/*)
```

harshad16 commented on ci-operator/config/opendatahub-io/notebooks/opendatahub-io-notebooks-main.yaml:
```diff
@@ -721,7 +721,7 @@ tests:
     workflow: opendatahub-io-ci-image-mirror
 - as: notebook-cuda-c9s-python-3-9-pr-image-mirror
-  skip_if_only_changed: ^docs/|\\.(md)$|^(\.gitignore|LICENSE|OWNERS|PROJECT)$
+  run_if_changed: ^(cuda/c9s-python-3.9/*)
```
```suggestion
  run_if_changed: ^(base/c9s-python-3.9/*)|(cuda/c9s-python-3.9/*)
```

harshad16 reviewed (changes requested):
The changes looks good to me.
i have dropped few suggestions.

also to be inline with regex, lets include to escape character i.e
a backlash \ infornt of forward / : like \/

atheo89 commented:
> The changes looks good to me. i have dropped few suggestions.

Oh right, thank you for noticing, I have updated them! :) 

> also to be inline with regex, lets include to escape character i.e a backlash \ infornt of forward / : like /

Added.

atheo89 commented:
pj-rehearse pull-ci-opendatahub-io-notebooks-main-notebooks-e2e-tests

atheo89 commented:
/pj-rehearse pull-ci-opendatahub-io-notebooks-main-notebooks-e2e-tests

harshad16 reviewed (commented):
/lgtm
/approve
thanks :100: 

the error from the ci check is not with this PR 
due to some other change in  repo 
as e2e contains conda changes.

harshad16 commented:
/pj-rehearse pull-ci-opendatahub-io-notebooks-main-notebook-base-c9s-python-3-9-pr-image-mirror

atheo89 commented:
/pj-rehearse skip pull-ci-opendatahub-io-notebooks-main-notebooks-e2e-tests

atheo89 commented:
/pj-rehearse skip

atheo89 commented:
/pj-rehearse abort pull-ci-opendatahub-io-notebooks-main-notebooks-e2e-tests

droslean commented:
/hold

Temporary holding to unblock the queue

droslean commented:
/hold

Temporary holding to unblock the queue

droslean commented:
/hold cancel
